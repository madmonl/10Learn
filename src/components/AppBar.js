import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Button, IconButton, Typography, Toolbar, AppBar, Tabs, Tab, Badge } from '@material-ui/core/';
import MenuIcon from '@material-ui/icons/Menu';
import { startLogout } from '../actions/auth';
import { connect } from 'react-redux';
import compose from 'recompose/compose';
import styled from 'styled-components';
import { PowerSettingsNew, Folder, Edit } from '@material-ui/icons';

// <div className={classes.container}>
const ToolbarCSS = styled(Toolbar)`
  display: flex;
  justify-content: space-between;
`;

const styles = {
  button: {
    size: 70,
    width: 100,
    fontSize: 15
  }
};

export class ButtonAppBar extends Component {
  constructor(props) {
     super(props);
  }
  logout = () => {
    if (this.props.guest) {
        this.props.dispatchGuestLogout();
    }
    else {
        this.props.startLogout();
    }
}

  render () {
    const { classes } = this.props;
    return (
        <div>
            <AppBar position="static" color="default">
              <div className="app-bar-header__theme">            
                <div className="content-container">
                  <ToolbarCSS>
                    <Typography color="textPrimary" variant="display2">
                        10Learn
                    </Typography>
                    <div className='app-bar-header'>
                      <Tabs value={0}>              
                        <Tab label={<span className="app-bar-header__tab">
                          <Edit />
                          תרגול
                        </span>} />
                        <Tab label={<span className="app-bar-header__tab">
                          <Folder />
                          מבחנים קודמים
                        </span>} />
                      </Tabs>
                      <Button className={classes.button} size='medium' onClick={this.logout}>
                        <PowerSettingsNew />
                        התנתק
                      </Button>
                    </div>          
                  </ToolbarCSS>
                </div>                
              </div>
            </AppBar>
        </div>
    );
  };
}

ButtonAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout()),
    dispatchGuestLogout: () => dispatch(dispatchGuestLogout())
});

const mapStateToProps = (state) => ({
    guest: state.auth.guest
});

export default compose(
    withStyles(styles), 
    connect(mapStateToProps, mapDispatchToProps)
  )(ButtonAppBar);