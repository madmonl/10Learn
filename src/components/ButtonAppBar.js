import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Button, IconButton, Typography, Toolbar, AppBar, Tabs, Tab, Badge, Paper } from '@material-ui/core/';
import MenuIcon from '@material-ui/icons/Menu';
import { startLogout, dispatchGuestLogout } from '../actions/auth';
import { connect } from 'react-redux';
import compose from 'recompose/compose';
import styled from 'styled-components';
import { PowerSettingsNew, Folder, Edit } from '@material-ui/icons';
import { dispatchHeaderTabsNavigation } from '../actions/material'

// <div className={classes.container}>
const ToolbarCSS = styled(Toolbar)`
  display: flex;
  justify-content: space-between;
`;

const TypographyTABS = styled(Typography)`
  display: flex;
  justify-content: center;
  margin-right: 20px;
`;

const TypographyCSS = styled(Typography)`
  font: italic bold Georgia;
`;

const styles = theme => ({
  button: {
    size: 70,
    width: 100,
    fontSize: 15
  }
});

export class ButtonAppBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      appBarTabsIndex: 0
    }
    this.logout = this.logout.bind(this);
    this.onIndexSelect = this.onIndexSelect.bind(this);
  }
  
  logout = () => {
    if (this.props.guest) {
        this.props.dispatchGuestLogout();
    }
    else {
        this.props.startLogout();
    }
  }

  onIndexSelect = (e, index) => {
    this.props.dispatchHeaderTabsNavigation(this.props.tabs[index]);
  }

  render () {
    const { classes, tabs, currTab } = this.props;
    const currTabIndex = tabs.indexOf(currTab);
    return (
        <div>
            <AppBar position="static" color="default">
              <div className="app-bar-header__theme">            
                <div className="content-container">
                  <ToolbarCSS>
                    <TypographyCSS color="textPrimary" variant="display2">
                        10Learn
                    </TypographyCSS>
                    <div className='app-bar-header'>
                      <Tabs 
                        value={currTabIndex}
                        indicatorColor="secondary"
                        onChange={this.onIndexSelect}
                      >
                        <Tab label={<span className="app-bar-header__tab">
                          <Edit className="app-bar-header__icon" />
                          <TypographyTABS className="app-bar-header__text" color="textPrimary" variant="title">
                            תרגול
                          </TypographyTABS>
                        </span>} />
                        <Tab label={<span className="app-bar-header__tab">
                          <Folder className="app-bar-header__icon" />
                          <TypographyTABS className="app-bar-header__text" color="textPrimary" variant="title">
                            מבחנים קודמים
                          </TypographyTABS>
                        </span>} />
                      </Tabs>
                      <Button className={classes.button} size='medium' onClick={this.logout}>
                        <PowerSettingsNew />
                        התנתק
                      </Button>
                    </div>       
                  </ToolbarCSS>
                </div>                
              </div>
            </AppBar>
        </div>
    );
  };
}

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout()),
    dispatchGuestLogout: () => dispatch(dispatchGuestLogout()),
    dispatchHeaderTabsNavigation: (currTab) => dispatch(dispatchHeaderTabsNavigation(currTab))
});

const mapStateToProps = (state) => ({
    guest: state.auth.guest,
    currTab: state.material.currTab,
    tabs: state.material.tabs
});

export default compose(
    withStyles(styles), 
    connect(mapStateToProps, mapDispatchToProps)
  )(ButtonAppBar);