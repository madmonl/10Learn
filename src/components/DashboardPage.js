import React, { Fragment } from 'react';
import LoadingPage from './LoadingPage';
import { Paper } from '@material-ui/core';
import PrevExams from './PrevExams';
import Practice from './Practice';
import { connect } from 'react-redux';

export const DashboardPage = ({ currTab }) => (
  <div>
    {currTab === 'תרגול' && <Practice/>}
    {currTab === 'מבחנים קודמים' && <PrevExams/>}
  </div>
);


const mapStateToProps = (state) => ({
    currTab: state.material.currTab
});

export default connect(mapStateToProps)(DashboardPage);
  
