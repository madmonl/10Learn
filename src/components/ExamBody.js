import React, { Fragment, Component } from 'react';
import MathJax from 'react-mathjax';
import { dispatchChangeQuestion, dispatchMarkQuestion, dispatchSetAnswer } from '../actions/exam';
import { connect } from 'react-redux';

export class ExamBody extends Component {
  constructor (props) {
    super(props);
    this.solutionClassName = this.solutionClassName.bind(this);
    this.onSubmitQuestion = this.onSubmitQuestion.bind(this);
  }

  onSubmitQuestion = (index, answer) => {
    this.props.dispatchChangeQuestion((index + 1) % 10);
    this.props.dispatchMarkQuestion(index);
    this.props.dispatchSetAnswer(index, answer);
  } 
  
  solutionClassName = (index, questions, currQuestion, questionsStatus, answersStatus) => {
    // one of:
    // "exam__answer-item" when still in exam - clickable.
    // "exam__answer-correct" not clickable - correct solution only.
    // "exam__answer-mistake" only a mistaken solution that was chosen.
    // or exam finished and
    // "exam__answer-default" not clickable not chosen, not correct answer.
    if (questionsStatus[currQuestion] === 'being_answered') {
      return "exam__answer-item"
    } else if (questions[currQuestion].index === index) {
      return "exam__answer-correct exam__answer-item"
    } else if (questions[currQuestion].index !== index 
      && answersStatus[currQuestion] === index) {
      return "exam__answer-mistake exam__answer-item"
    } else {
      return "exam__answer-default exam__answer-item"
    };
  }
  
  render () {
    const { questions, currQuestion, questionsStatus, answersStatus } = this.props;
    
    return (
      <div>
        <div className="exam__question-navigation-container">
          <div className="exam__question">
            <MathJax.Provider>
              <div>
                  <MathJax.Node formula={questions[currQuestion].question} />
              </div>
            </MathJax.Provider>
          </div>
        </div>
        <div className="exam__answers">
          {questions[currQuestion].solutions.map((solution, index) =>
            <div 
              className={
                this.solutionClassName(index, questions, currQuestion, questionsStatus, answersStatus)
              }
              key={index}
              onClick={() => this.onSubmitQuestion(currQuestion, index)}
            >
              <MathJax.Provider>
                <div>
                    <MathJax.Node formula={solution} />
                </div>
              </MathJax.Provider>
            </div>             
          )}
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
    dispatchChangeQuestion: (question) => dispatch(dispatchChangeQuestion(question)),
    dispatchMarkQuestion: (index) => dispatch(dispatchMarkQuestion(index)),
    dispatchSetAnswer: (index, answer) => dispatch(dispatchSetAnswer(index, answer))
});

const mapStateToProps = (state) => ({
    questions: state.exam.questions,
    currQuestion: state.exam.currQuestion,
    questionsStatus: state.exam.questionsStatus,
    answersStatus: state.exam.answersStatus,
});
  
export default connect(mapStateToProps, mapDispatchToProps)(ExamBody);
