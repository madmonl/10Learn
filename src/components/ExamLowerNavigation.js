
import React, { Fragment, Component } from 'react';
import { dispatchChangeQuestion } from '../actions/exam';
import { Button } from '@material-ui/core/';
import { withStyles } from '@material-ui/core/styles';
import { Send } from '@material-ui/icons';
import { connect } from 'react-redux';
import compose from 'recompose/compose';

export const styles = theme => ({
  button: {
    margin: theme.spacing.unit
  },
  rightIcon: {
    marginLeft: theme.spacing.unit
  },
  leftIcon: {
    marginLeft: theme.spacing.unit
  }
})

export class ExamLowerNavigation extends Component {
  constructor (props) {
    super(props);
    this.onQuestionChange = this.onQuestionChange.bind(this);
  }
  
  onQuestionChange = direction => {
    if (direction === 'prev'){
      this.props.dispatchChangeQuestion( this.props.currQuestion === 0 
        ? this.props.questions.length - 1
        : this.props.currQuestion - 1)
    } else if (direction === 'next') {
      this.props.dispatchChangeQuestion((this.props.currQuestion + 1) % this.props.questions.length)
    } else {
      throw "Button direction is incorrect.";
    }
  }

  render () {
    const { questions, currQuestion, classes } = this.props;
    
    return (
      <div>
        <div className="exam__navigation-container">
          <Button 
            variant="contained" 
            color="primary" 
            className={classes.button}
            onClick={() => this.onQuestionChange('prev')}
          >
            <Send className={classes.rightIcon} />
            שאלה קודמת            
          </Button>
          <Button 
            variant="contained" 
            color="primary" 
            className={classes.button}
            onClick={() => this.onQuestionChange('next')}
          >
            שאלה הבאה
            <Send className={classes.leftIcon} />
          </Button>
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
    dispatchChangeQuestion: (question) => dispatch(dispatchChangeQuestion(question))
});

const mapStateToProps = (state) => ({
    questions: state.exam.questions,
    currQuestion: state.exam.currQuestion
});
  
export default compose(
  withStyles(styles), 
  connect(mapStateToProps, mapDispatchToProps))
  (ExamLowerNavigation);

    