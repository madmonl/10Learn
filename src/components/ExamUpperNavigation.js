import React, { Component } from 'react';
import NumbersNavigation from './NumbersNavigation';
import SubmitExamModal from './SubmitExamModal';




export default () =>
  <div className="exam__upper-footer">
    <NumbersNavigation />
    <SubmitExamModal />
  </div>

