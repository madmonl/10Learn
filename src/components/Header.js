import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout, dispatchGuestLogout } from '../actions/auth';

export class Header extends React.Component {
    constructor (props) {
        super(props);
        this.logout = this.logout.bind(this);
    }

    logout () {
        if (this.props.guest) {
            this.props.dispatchGuestLogout();
        }
        else {
            this.props.startLogout();
        }
    }
    
    render () {
        return (
            <header className="header">
                <div className="content-container">
                    <div className="header__content">
                        <Link className="header__title" to="/dashboard">
                            <h1>Boilerplate</h1>
                        </Link>
                        <button className="button button--link" onClick={this.logout}>Logout</button>
                    </div>
                </div>
            </header>
        );
    }
};

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout()),
    dispatchGuestLogout: () => dispatch(dispatchGuestLogout())
});

const mapStateToProps = (state) => ({
    guest: state.auth.guest
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
