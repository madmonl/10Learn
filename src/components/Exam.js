import React, { Fragment, Component } from 'react';
import ExamUpperNavigation from './ExamUpperNavigation';
import ExamBody from './ExamBody';
import ExamLowerNavigation from './ExamLowerNavigation';

export default () => 
  <div className="exam">
    <ExamUpperNavigation />
    <ExamBody />
    <ExamLowerNavigation />        
  </div>

