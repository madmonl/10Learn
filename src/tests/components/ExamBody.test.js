import React from 'react';
import { shallow } from 'enzyme';
import { ExamBody } from '../../components/ExamBody';
import exams from '../fixtures/exams'

let wrapper, dispatchChangeQuestion, dispatchMarkQuestion, dispatchSetAnswer;

beforeEach(() => {
  dispatchChangeQuestion = jest.fn();
  dispatchMarkQuestion = jest.fn();
  dispatchSetAnswer = jest.fn();
  wrapper = shallow(
    <ExamBody 
      dispatchChangeQuestion={dispatchChangeQuestion}
      dispatchMarkQuestion={dispatchMarkQuestion}
      dispatchSetAnswer={dispatchSetAnswer}
      questions={exams[0].questions} 
      currQuestion={0}
      questionsStatus={[]}
      answersStatus={[]}
    />
  );
});

test('should render ExamBody correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should call solutions` on click correctly', () => {
  wrapper.find('.exam__answers').children().forEach((node) => {
    node.prop('onClick')();
  });
  expect(dispatchChangeQuestion).toHaveBeenCalledTimes(3);
  expect(dispatchMarkQuestion).toHaveBeenCalledTimes(3);
  expect(dispatchSetAnswer).toHaveBeenCalledTimes(3);
});