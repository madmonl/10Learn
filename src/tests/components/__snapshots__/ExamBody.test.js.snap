// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExamBody correctly 1`] = `
<div>
  <div
    className="exam__question-navigation-container"
  >
    <div
      className="exam__question"
    >
      <MathJaxProvider
        options={
          Object {
            "showMathMenu": false,
            "showMathMenuMSIE": false,
            "tex2jax": Object {
              "inlineMath": Array [],
            },
          }
        }
        script="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML"
      >
        <div>
          <MathJaxNode
            formula="1 + 2 = ?"
          />
        </div>
      </MathJaxProvider>
    </div>
  </div>
  <div
    className="exam__answers"
  >
    <div
      className="exam__answer-default exam__answer-item"
      key="0"
      onClick={[Function]}
    >
      <MathJaxProvider
        options={
          Object {
            "showMathMenu": false,
            "showMathMenuMSIE": false,
            "tex2jax": Object {
              "inlineMath": Array [],
            },
          }
        }
        script="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML"
      >
        <div>
          <MathJaxNode
            formula={1}
          />
        </div>
      </MathJaxProvider>
    </div>
    <div
      className="exam__answer-correct exam__answer-item"
      key="1"
      onClick={[Function]}
    >
      <MathJaxProvider
        options={
          Object {
            "showMathMenu": false,
            "showMathMenuMSIE": false,
            "tex2jax": Object {
              "inlineMath": Array [],
            },
          }
        }
        script="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML"
      >
        <div>
          <MathJaxNode
            formula={3}
          />
        </div>
      </MathJaxProvider>
    </div>
    <div
      className="exam__answer-default exam__answer-item"
      key="2"
      onClick={[Function]}
    >
      <MathJaxProvider
        options={
          Object {
            "showMathMenu": false,
            "showMathMenuMSIE": false,
            "tex2jax": Object {
              "inlineMath": Array [],
            },
          }
        }
        script="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML"
      >
        <div>
          <MathJaxNode
            formula={4}
          />
        </div>
      </MathJaxProvider>
    </div>
  </div>
</div>
`;
